server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ReactorNettyJaxrsServer.java

```diff
@@ -369,26 +373,41 @@ private SslContext toNettySSLContext(final SSLContext sslContext)
       );
    }
 
-   private ResteasyUriInfo extractUriInfo(final HttpServerRequest req, final String contextPath)
-   {
-      final String uri = req.uri();
-
-      String uriString;
-
-      // If we appear to have an absolute URL, don't try to recreate it from the host and request line.
-      if (uri.startsWith(req.scheme() + "://")) {
-         uriString = uri;
-      } else {
-         uriString = new StringBuilder(100)
-             .append(req.scheme())
-             .append("://")
-             .append(req.hostAddress().getHostString())
-             .append(":").append(req.hostAddress().getPort())
-             .append(req.uri())
-             .toString();
+   static class UriExtractor {
+      ResteasyUriInfo extract(final HttpServerRequest req, final String contextPath)
+      {
+         final String uri = req.uri();
+
+         final String uriString;
+
+         // If we have an absolute URL, don't try to recreate it from the host and request line.
+         if (uri.startsWith(req.scheme() + "://")) {
+            uriString = uri;
+         } else {
+            String host = req.requestHeaders().get(HttpHeaderNames.HOST);
+            if (host == null || "".equals(host.trim())) {
+               final InetSocketAddress hostAddress = req.hostAddress();
+               if (hostAddress == null) {
```

Comment:
What do you think @jamezp , this or:
```
assert hostAddress != null : "Host address not accessible from request";
```
Also, a netty4 adapter test failed in CI.  I hope that's not related to my change!

Reply:
My only preference for the `IllegalArgumentException` is just because assertions aren't usually used in production.

> Also, a netty4 adapter test failed in CI. I hope that's not related to my change!
Nah, that test is just flaky for some reason. I've been meaning to look at it, but haven't found the time.

Reply:
Makes sense to me.  Is there anything else I need to do to get this merged?

Reply:
Sorry, I meant a PR for main as well :)